### DISCUSSION TITLE
Visual search for better quality duplicate

### DISCUSSION THREAD

**Discussion by sazarkin** [photoprism/photoprism#4127] [Q&A] [answered]
I have a folder of photos that I intend to print, but some of them are of lower quality than desired. I'm seeking a method to search for higher-quality originals within my Photoprism library using these lower-quality files.
Is this feasible?

**Comment by lastzero** [comment, marked as answer]
You may be able to find them by sorting by visual similarity or filtering by date:
- https://demo.photoprism.app/library/browse?view=cards&order=similar&public=true&quality=3

There is currently no "button" you can click to find similar pictures only. However, you could (manually) use the `diff:...` search filter, which filters by the differential perception hash generated during indexing (a hexadecimal value between 000000 and FFFFFF; I don't know if it's displayed anywhere in the UI yet, otherwise you'll have to check the files database table for it or use the browser developer tools to inspect the server response):
- https://www.photoprism.app/kb/search-filters

**Comment by sazarkin** [comment]
Filtering by the "diff" attribute did not yield sufficiently specific results, as it often returned approximately 4,000 matches for some images. Consequently, I resorted to utilizing a VGG embedding from the repository found at https://github.com/christiansafka/img2vec. I wrote a Python script to search for similarity on top of phtotoprism collections

**Reply by lastzero** [reply]
Yes, the main purpose was to be able to use this diff to sort pictures in combination with other attributes such as color.

There are **better ways to find duplicates**, but unfortunately we haven't had time to compare them in terms of performance and usefulness, let alone implement them in native Go (since we don't want to run a Python script or external service for each file during indexing):
- https://docs.photoprism.app/developer-guide/metadata/perceptual-hashes/

Any help with that would be much appreciated! 💎
